src/components/ChatEntry.js

```diff
@@ -1,22 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
+import TimeStamp from './TimeStamp';
 import PropTypes from 'prop-types';
 
 const ChatEntry = (props) => {
+  // const onLike = () => {
+  //   props.updateLike(props.id);
+  // }
+
+  const [isLiked, setIsLiked] = useState(false);
+
+  const likeFill = isLiked ? '❤️' : '🤍';
+
+  const toggleLiked = () => {
+    setIsLiked(!isLiked);
+  };
+
+  const onLike = () => {
+    props.updateLike(props.id);
+    toggleLiked();
+  };
+
+  // This line of code is defining a function called onLike. When this function
+  //is called, it will call the updateLike function, passing in the props.id as an argument.
+  // The updateLike function is likely a function that updates the like status of the item with the given id.
+  // The onLike function will be used later in the code as an event handler for a button or some other UI element
+  //that allows the user to like or unlike an item.
+
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp} />
+        </p>
+        {/* <button className="like">🤍</button> */}
+        <button className="like" onClick={onLike}>
+          {likeFill}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
```

Comment:
Here you would want to have a `propType` for every prop passed in, therefore you would want one for all of the ones passed in `chatLog.js`